atlasdb-perf/src/main/java/com/palantir/atlasdb/performance/benchmarks/SweepBenchmarks.java

```diff
@@ -33,13 +33,66 @@
     @Measurement(time = 5, timeUnit = TimeUnit.SECONDS)
     public Object singleSweepRun(RegeneratingTable.SweepRegeneratingTable table) {
         SweepTaskRunner sweepTaskRunner = table.getSweepTaskRunner();
         SweepResults sweepResults = sweepTaskRunner.run(table.getTableRef(), 1, null);
-        assertThat(sweepResults.getCellsDeleted(), is(99L));
+        assertThat(sweepResults.getCellsDeleted(), is(DELETED_COUNT));
+        return sweepResults;
+    }
+
+    @Benchmark
+    @Warmup(time = 2, timeUnit = TimeUnit.SECONDS)
+    @Measurement(time = 10, timeUnit = TimeUnit.SECONDS)
+    public Object batchedUniformSingleSweepRun(RegeneratingTable.SweepBatchUniformMultipleRegeneratingTable table) {
+        SweepTaskRunner sweepTaskRunner = table.getSweepTaskRunner();
+        SweepResults sweepResults = sweepTaskRunner.run(table.getTableRef(), BATCH_SIZE, null);
+        assertThat(sweepResults.getCellsDeleted(), is(DELETED_COUNT * BATCH_SIZE));
+        return sweepResults;
+    }
+
+    @Benchmark
+    @Warmup(time = 6, timeUnit = TimeUnit.SECONDS)
+    @Measurement(time = 30, timeUnit = TimeUnit.SECONDS)
+    public Object batchedSingleSweepRun(RegeneratingTable.SweepBatchNonUniformMultipleSeparateRegeneratingTable table) {
+        SweepTaskRunner sweepTaskRunner = table.getSweepTaskRunner();
+        SweepResults sweepResults = sweepTaskRunner.run(table.getTableRef(), BATCH_SIZE, null);
+        assertThat(sweepResults.getCellsDeleted(), is(DELETED_COUNT * BATCH_SIZE));
+        return sweepResults;
+    }
+
+    @Benchmark
+    @Warmup(time = 3, timeUnit = TimeUnit.SECONDS)
+    @Measurement(time = 15, timeUnit = TimeUnit.SECONDS)
+    public Object multipleUniformSweepRun(RegeneratingTable.SweepBatchUniformMultipleRegeneratingTable table) {
+        SweepTaskRunner sweepTaskRunner = table.getSweepTaskRunner();
+        SweepResults sweepResults = null;
+        byte[] nextStartRow = null;
+        for (int i = 0; i < BATCH_SIZE; i++){
+            sweepResults = sweepTaskRunner.run(table.getTableRef(), 1, nextStartRow);
+            nextStartRow = sweepResults.getNextStartRow().get();
+            assertThat(sweepResults.getCellsDeleted(), is(DELETED_COUNT));
+        }
+        return sweepResults;
+    }
+
+    @Benchmark
+    @Warmup(time = 8, timeUnit = TimeUnit.SECONDS)
+    @Measurement(time = 40, timeUnit = TimeUnit.SECONDS)
+    public Object multipleSweepRun(RegeneratingTable.SweepBatchNonUniformMultipleSeparateRegeneratingTable table) {
+        SweepTaskRunner sweepTaskRunner = table.getSweepTaskRunner();
+        SweepResults sweepResults = null;
+        byte[] nextStartRow = null;
+        for (int i = 0; i < BATCH_SIZE; i++){
+            sweepResults = sweepTaskRunner.run(table.getTableRef(), 1, nextStartRow);
+            nextStartRow = sweepResults.getNextStartRow().get();
+            assertThat(sweepResults.getCellsDeleted(), is(DELETED_COUNT));
+        }
         return sweepResults;
     }
```

Comment:
you can reduce the duplicated code here by having a generic method that takes a RegeneratingTable (abstract type) that backs both flavors of each test (see KvsGetRangeBenchmarks.java for an example)